### PR TITLE
geoipupdate: Init at 4.1.5

### PR DESCRIPTION
**Pull request by dasJ**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by dasJ** [comment]
See also #55867

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ buildGoModule, fetchFromGitHub, stdenv }: buildGoModule rec {
+  pname = "geoipupdate";
+  version = "4.1.5";
+
+  src = fetchFromGitHub {
+    owner = "maxmind";
+    repo = "geoipupdate";
+    rev = "v${version}";
+    sha256 = "sha256:1k0bmsqgw35sdmaafinlr4qd5910fi598i8irxrz11394d3c8giv";
```
Comment (prusnak) [inline pkgs/applications/misc/geoipupdate/default.nix]: Drop the `sha256:` prefix

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  };
+
+  modSha256 = "sha256:0jlg2p5qzfrw64ay5fphigf36yf67rjyw05vj5xhkqa630ph5ydr";
```
Comment (prusnak) [inline pkgs/applications/misc/geoipupdate/default.nix]: Drop the `sha256:` prefix

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{ buildGoModule, fetchFromGitHub, stdenv }: buildGoModule rec {
```
Comment (prusnak) [inline pkgs/applications/misc/geoipupdate/default.nix]: Let's use this order: `{ stdenv, buildGoModule, fetchFromGitHub }:` and add a newline after `:`.

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by dasJ** [comment]
@prusnak Done

**Comment by prusnak** [comment]
@GrahamcOfBorg build geoipupdate

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    repo = "geoipupdate";
+    rev = "v${version}";
+    sha256 = "1k0bmsqgw35sdmaafinlr4qd5910fi598i8irxrz11394d3c8giv";
+  };
+
+  modSha256 = "0jlg2p5qzfrw64ay5fphigf36yf67rjyw05vj5xhkqa630ph5ydr";
+
+  meta = with stdenv.lib; {
+    description = "Automatic GeoIP database updater";
+    license = with licenses; [ asl20 ];
```
Comment (prusnak) [inline pkgs/applications/misc/geoipupdate/default.nix]: Sorry, one more :-/

`    homepage = "https://github.com/maxmind/geoipupdate"`;

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by dasJ** [comment]
Hash mismatches were due to me using the wrong callPackage while testing…

**Comment by dasJ** [comment]
@GrahamcOfBorg build geoipupdate

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (jonringer) [inline pkgs/applications/misc/geoipupdate/default.nix]: common to have a newline here
```suggestion
{ stdenv, buildGoModule, fetchFromGitHub }:

```

**Comment by dasJ** [comment]
@jonringer Done

File: pkgs/applications/misc/geoipupdate/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "v${version}";
+    sha256 = "1k0bmsqgw35sdmaafinlr4qd5910fi598i8irxrz11394d3c8giv";
+  };
+
+  modSha256 = "0mk6zp6byq3jc6wipx53bg5igry114klq5w8isc0z6r63zjsk6f6";
+
+  meta = with stdenv.lib; {
+    description = "Automatic GeoIP database updater";
+    license = with licenses; [ asl20 ];
+    platforms = platforms.all;
```
Comment (veprbl) [inline pkgs/applications/misc/geoipupdate/default.nix]: This line is not needed:
https://github.com/NixOS/nixpkgs/blob/74edb58ad19225222a72c14bb37f3b42e5f0a344/pkgs/development/go-modules/generic/default.nix#L200-L201

**Comment by dasJ** [comment]
@veprbl Done

**Review by jonringer — APPROVED** [review]
LGTM, please apply @prusnak suggestion :)

**Comment by dasJ** [comment]
@jonringer what do you mean? Have I overlooked something?

File: pkgs/applications/misc/geoipupdate/default.nix (hunk 4 above)
Comment (dasJ) [reply pkgs/applications/misc/geoipupdate/default.nix]: I thought about that but I wasn't sure if this is really the homepage :/

**Comment by dasJ** [comment]
@prusnak @jonringer Done

File: pkgs/applications/misc/geoipupdate/default.nix (hunk 4 above)
Comment (jonringer) [reply pkgs/applications/misc/geoipupdate/default.nix]: It's just some way to link to the correct repo, as there may be other repos with the same name. github is commonly used as a landing page

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
commits LGTM
repo looks active enough

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73578
1 package were build:
geoipupdate
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build geoipupdate

**Pull request merged**